internal/resource/url.go

```diff
@@ -432,8 +432,7 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 	}
 	if opts.Hash != nil {
 		opts.Hash.Reset()
-		_, err = dest.Seek(0, os.SEEK_SET)
-		if err != nil {
+		if _, err = dest.Seek(0, io.SeekStart); err != nil {
```

Comment:
The two separate lines are better in this case, because this line is misleading.  The compound `if` is usually used to create a new `err` variable, but here we're reusing one that already exists.